core-container/entry.sh

```diff
@@ -16,88 +16,402 @@
 
 set -e
 
-echo "< Starting Kubestellar container >-------------------------"
 
-# Try to guess the route/ingress
-if [ -z "$EXTERNAL_HOSTNAME" ]; then
-    if kubectl get route kubestellar-route &> /dev/null; then
-        EXTERNAL_HOSTNAME=$(kubectl get route kubestellar-route -o yaml -o jsonpath={.spec.host} 2> /dev/null)
+espw_name="espw"
+
+
+function echoerr() {
+   echo "ERROR: $1" >&2
+}
+
+
+function wait_kcp_ready() {
+    echo "Waiting for kcp to be ready... this may take a while."
+    (
+        KUBECONFIG=
+        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp | grep '***READY***')" != "" ]; do
+            sleep 5
+        done
+    )
+    echo "Succes!"
+    echo "Copying the admin.kubeconfig from kubestellar seret..."
+    mkdir -p /home/kubestellar/.kcp
+    (
+        KUBECONFIG=
+        kubectl get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > /home/kubestellar/.kcp/admin.kubeconfig
+    )
```

Comment:
This seems more complicated than necesssary. Why not just mount the persistent volume at `home/kubestellar/.kcp` for every container?